plugin/trino-mariadb/src/main/java/io/trino/plugin/mariadb/MariaDbClient.java

```diff
@@ -484,10 +493,18 @@ protected void copyTableSchema(Connection connection, String catalogName, String
     }
 
     @Override
-    protected String createTableSql(RemoteTableName remoteTableName, List<String> columns, ConnectorTableMetadata tableMetadata)
+    protected void createTable(Connection connection, RemoteTableName remoteTableName, List<String> columns, ConnectorTableMetadata tableMetadata)
     {
         checkArgument(tableMetadata.getProperties().isEmpty(), "Unsupported table properties: %s", tableMetadata.getProperties());
-        return format("CREATE TABLE %s (%s) COMMENT '%s'", quoted(remoteTableName), join(", ", columns), tableMetadata.getComment().orElse(NO_COMMENT));
+        String sql = format("CREATE TABLE %s (%s) COMMENT ?", quoted(remoteTableName), join(", ", columns));
```

Comment:
I am not sure there is much benefit of using the prepared statement for the comment.
it's nice, but we still need to know how to escape all the other stuff (table name, column names, check constraints (in the future), etc). Escaping the COMMENT value isn't any harder that what we do today.

Reply:
In case of MariaDB, they have `NO_BACKSLASH_ESCAPES` (global or session) property. If we escape the comment without respecting the property in our side, the inserted comment might be different from what we specified in Trino DDL. Also, I guess there're other cases we are missing. That's why I introduced prepared statements in this PR.

Reply:
We control the MariaDB connection (session), so we should force NO_BACKSLASH_ESCAPES to be true.
This is important for complex predicate pushdown as well, see `RewriteVarcharConstant`.

Reply:
Reverted prepared statement change. Do you want me to force NO_BACKSLASH_ESCAPES in this PR?

Reply:
We can do this as a follow up (It's a "minor correctness" problem. We just don't properly support comments with backslashes.)